src/sage/schemes/weighted_projective/weighted_projective_point.py

```diff
@@ -0,0 +1,297 @@
+
+    - ``check`` -- boolean (default: ``True``). Whether to check the input for consistency.
+
+    EXAMPLES::
+
+        sage: WP = WeightedProjectiveSpace(2, ZZ)
+        sage: WP(2, 3, 4)
+        (2 : 3 : 4)
+    """
+
+    def __init__(self, X, v, check=True):
+        """
+        The Python constructor.
+
+        EXAMPLES::
+
+            sage: WP = WeightedProjectiveSpace([3, 4, 5], QQ)
+            sage: P1 = WP(2, 3, 4); P1
+            (2 : 3 : 4)
+            sage: P2 = WP(2000, 30000, 400000); P2
+            (2000 : 30000 : 400000)
+            sage: P1 == P2
+            True
+
+        The point constructor normalises coordinates at the last position with
+        weight `1`::
+
+            sage: WP = WeightedProjectiveSpace([3, 1, 4, 1], QQ)
+            sage: P = WP(8, 3, 16, 2); P
+            (1 : 3/2 : 1 : 1)
+            sage: P == WP(1, 3 / 2, 1, 1)
+            True
+        """
+        SchemeMorphism.__init__(self, X)
+
+        self._normalized = False
+
+        if check:
+            # check parent
+            from sage.schemes.weighted_projective.weighted_projective_homset import SchemeHomset_points_weighted_projective_ring
+            if not isinstance(X, SchemeHomset_points_weighted_projective_ring):
+                raise TypeError(f"ambient space {X} must be a weighted projective space")
+
+            from sage.rings.ring import CommutativeRing
+            d = X.codomain().ambient_space().ngens()
+            if isinstance(v, SchemeMorphism):
+                v = list(v)
+            else:
+                try:
+                    if isinstance(v.parent(), CommutativeRing):
+                        v = [v]
+                except AttributeError:
+                    pass
+            if not isinstance(v, (list, tuple)):
+                raise TypeError("argument v (= %s) must be a scheme point, list, or tuple" % str(v))
+            if len(v) != d and len(v) != d-1:
+                raise TypeError("v (=%s) must have %s components" % (v, d))
+
+            R = X.value_ring()
+            v = Sequence(v, R)
+            if len(v) == d-1:     # very common special case
+                v.append(R.one())
+
+            # (0 : 0 : ... : 0) is not a valid (weighted) projective point
+            if not any(v):
+                raise ValueError(f"{v} does not define a valid projective "
+                                 "point since all entries are zero")
+
+            # over other rings, we do not have a generic method to check
+            # whether the given coordinates is a multiple of a zero divisor
+            # so we just let it pass.
+            X.extended_codomain()._check_satisfies_equations(v)
+
+            self._coords = tuple(v)
+
+            # we (try to) normalise coordinates!
+            self.normalize_coordinates()
+
+        else:
+            self._coords = tuple(v)
+
+    def _repr_(self):
+        return "({})".format(" : ".join(map(repr, self._coords)))
```

Comment:
```suggestion
        return "({' : '.join(map(repr, self._coords))})"
```

If you want. I usually find f-strings more readable but this is a case where either is fine. I think f-strings used to be faster but that's no longer the case, I think they're the same speed now.

Reply:
I decide not to. The part inside is too long.